README-distccd.md

```diff
@@ -0,0 +1,124 @@
+```
+`172.27.124.34` can be replaced with server IP and `172.27.0.0/16` with subnet you want to allow connection from.
+
+The above setup is good enough when dealing over local networking. For servers on remote location following setup can be considered:
+
+### Zerotier Setup
+
+I found this to be the best and quickest way to setup a Virtual Private Network for this purpose.
+
+* Create an account at https://www.zerotier.com
+* Setup on both host and server https://www.zerotier.com/download/
+* Setup Network on your account
+* Join the network on each server and host using network id
+* You will need to accept the accounts and give the name under members list.
+
+For more info please refer Zerotier docs.
+
+## Host Side setup
+
+Following steps assume that you have ccache configured and tested working.
+
+* macos
+
+    * Install distcc using `brew install distcc`
+    * For building from source ensure you run `./configure --without-libiberty` at configure step.
+    * add following lines to your `~/.bash_profile`
+    ```
+    export DISTCC_HOSTS="<hostname or address>/<max number of jobs>,cpp,lzo"
+    export CCACHE_PREFIX="distcc"
+    ```
+    * `DISTCC_HOSTS` is space separated list of supported servers.
+    * you can add hostnames in `/etc/hosts` files for better management, the list shall contain all the servers setup using Server side setup.
+    * Run build using `./waf <build-target> -j$(distcc -j)`
+
+* ubuntu
+
+    * Install distcc using `sudo apt-get install distcc`
+    * add following lines to your `~/.bashrc`
+    ```
+    export DISTCC_HOSTS="<hostname or address>/<max number of jobs>,cpp,lzo"
+    export CCACHE_PREFIX="distcc"
+    ```
+    * `DISTCC_HOSTS` is space separated list of supported servers.
+    * you can add hostnames in `/etc/hosts` files for better management, the list shall contain all the servers setup using Server side setup.
+    * For better monitoring a tool called `distccmon-gnome` can be installed `sudo apt-get install distccmon-gnome`
+    * Run build using `./waf <build-target> -j$(distcc -j)`
+
+## Additional Notes
+
+Following are the key points one can play with to come up with optimised setup:
+
+* If you want to push every part of compilation to remote server including preprocessing one can use `pump ./waf <build-target> -j$(distcc -j)`. This can be very handy when your local machine is very slow, and you have a significant bandwidth to the remote server.
+
+* For debugging on server side you can run the docker in interactive mode like this:
+`docker run -it -p 3632:3632 -p 3633:3633 ardupilot/ardupilot-dev-distccd --jobs <Max number of jobs>`
+
+* For debugging on host side simply `export DISTCC_VERBOSE=1`
+
+* There is no ccache on the server side docker, so as to limit the storage consumption on the remote machine. But your local ccache will still be in play even when using `pump`. If one wants, they can create a copy of docker with ccache setup, it should be trivial.
```

Comment:
why not use ccache on the docker image ? we could put a simple 1Go limit if needed